kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.stats;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.data.GenericRow;
+import io.delta.kernel.internal.util.InternalUtils;
+import io.delta.kernel.internal.util.VectorUtils;
+import io.delta.kernel.types.ArrayType;
+import io.delta.kernel.types.LongType;
+import io.delta.kernel.types.StructType;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/** A histogram that tracks file size distributions and their counts. */
+public class FileSizeHistogram {
+
+  //////////////////////////////////
+  // Static variables and methods //
+  //////////////////////////////////
+
+  private static final long KB = 1024;
+  private static final long MB = KB * 1024;
+  private static final long GB = MB * 1024;
+
+  public static final StructType FULL_SCHEMA =
+      new StructType()
+          .add("sortedBinBoundaries", new ArrayType(LongType.LONG, false))
+          .add("fileCounts", new ArrayType(LongType.LONG, false))
+          .add("totalBytes", new ArrayType(LongType.LONG, false));
+
+  /**
+   * Creates a default FileSizeHistogram with predefined bin boundaries and zero counts.
+   *
+   * @return A new FileSizeHistogram instance with default boundaries and zero counts
+   */
+  public static FileSizeHistogram createDefaultHistogram() {
+    long[] defaultBoundaries = createDefaultBinBoundaries();
+    long[] zeroCounts = new long[defaultBoundaries.length];
+    long[] zeroBytes = new long[defaultBoundaries.length];
+    return new FileSizeHistogram(defaultBoundaries, zeroCounts, zeroBytes);
+  }
+
+  /** Creates a FileSizeHistogram from a column vector. */
+  public static Optional<FileSizeHistogram> fromColumnVector(ColumnVector vector, int rowId) {
+    if (vector.isNullAt(rowId)) {
+      return Optional.empty();
+    }
+
+    int boundariesIdx = FULL_SCHEMA.indexOf("sortedBinBoundaries");
+    int totalBytesIdx = FULL_SCHEMA.indexOf("totalBytes");
+    int fileCountsIdx = FULL_SCHEMA.indexOf("fileCounts");
+
+    List<Long> boundariesList =
```

Comment:
How much work would it be to avoid creating the `List<Long>` and just create the `long[]` directly?

Reply:
I will add a followup to add toJavaArray in VectorUtils for this.